shared/src/main/java/dice/StatsProcessor.java

```diff
@@ -135,12 +135,124 @@ public ConcreteStatsProcessor(Database db) {
 
     public StatsData getPlayerStats(String playerName) {
         RollRecord[] records = database.getRecordsForUser(playerName);
+        boolean playerExists = false;
+        String[] playerList = getPlayerList();
+        int dbLength = records.length;
+        int totalRolls = dbLength;
+        int totalGames = 0;
+        int cumulativeScore = 0;
+        int totalDiceUsed = 0;
+        double avgRolls = 0.0;
+        double avgScore = 0.0;
+        double avgNumDiceUsed = 0.0;
+
+        // check to see if the Given player name exists
+        for (int index = 0; index < playerList.length; index++) {
+            if (playerList[index] == playerName) {
+                playerExists = true;
+            }
+        }
+
+        if (playerExists) {
```

Comment:
You should be able to check `records.length` here. If it's 0 then there's no data for the player